_posts/2023-01-22-auto-restarting-connectors.md

```diff
@@ -0,0 +1,312 @@
+java.lang.RuntimeException: Intentional task failure after receiving 5 records.
+	at cz.scholz.kafka.connect.echosink.EchoSinkTask.put(EchoSinkTask.java:131)
+	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:581)
+	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
+	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
+	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
+	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:189)
+	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:244)
+	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
+	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
+	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
+	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
+	at java.base/java.lang.Thread.run(Thread.java:833)
+```
+
+And the task fails.
+In the next reconciliation, the operator restarts the task.
+You can see this in the operator log:
+
+```
+2023-01-22 23:04:24 INFO  AbstractOperator:239 - Reconciliation #73(timer) KafkaConnect(myproject/my-connect): KafkaConnect my-connect will be checked for creation or modification
+2023-01-22 23:04:24 INFO  AbstractConnectOperator:502 - Reconciliation #73(timer) KafkaConnect(myproject/my-connect): creating/updating connector: echo-sink
+2023-01-22 23:04:24 INFO  AbstractConnectOperator:692 - Reconciliation #73(timer) KafkaConnect(myproject/my-connect): Auto restarting connector echo-sink
+2023-01-22 23:04:24 INFO  AbstractConnectOperator:696 - Reconciliation #73(timer) KafkaConnect(myproject/my-connect): Restarted connector echo-sink
+2023-01-22 23:04:24 INFO  CrdOperator:133 - Reconciliation #73(timer) KafkaConnect(myproject/my-connect): Status of KafkaConnector echo-sink in namespace myproject has been updated
+2023-01-22 23:04:24 INFO  AbstractOperator:510 - Reconciliation #73(timer) KafkaConnect(myproject/my-connect): reconciled
+```
+
+And you can see the status of the `KafkaConnector` resource updated as well:
+
+```yaml
+# ...
+status:
+  # ...
+  autoRestart:
+    count: 1
+    lastRestartTimestamp: "2023-01-22T23:04:24.386944356Z"
+```
+
+If you want, you can send another batch of messages to see it fail again, and check that with every restart the restart counter increases and takes longer before the next restart happens.
+Or you can stop sending messages.
+In that case, the connector and its task keeps running and the auto-restart counter eventually resets to 0 again.
```

Comment:
How long is the duration before the auto-restart counter is reset?

Reply:
Added more details on this.